graphql/client.go

```diff
@@ -239,3 +258,161 @@ func (c *client) createGetRequest(req *Request) (*http.Request, error) {
+	httpRequest, err := http.NewRequest(http.MethodPost, endpoint, http.NoBody)
+	if err != nil {
+		return nil, fmt.Errorf("error creating request: %w", err)
+	}
+	bodyBuf := &bytes.Buffer{}
+	bodyWriter := multipart.NewWriter(bodyBuf)
+	defer bodyWriter.Close()
+
+	// operations
+	requestBody, err := json.Marshal(req)
+	if err != nil {
+		return nil, fmt.Errorf("error marshalling request: %w", err)
+	}
+	err = bodyWriter.WriteField("operations", string(requestBody))
+	if err != nil {
+		return nil, fmt.Errorf("error writing operations to body: %w", err)
+	}
+
+	// map
+	mapData := ""
+	// group files to avoid uploading duplicated files
+	filesGroup := [][]*fileVariable{}
+	for _, file := range fileVariables {
+		foundDuplicated := false
+		for group, fileGroup := range filesGroup {
+			file2 := fileGroup[0]
+			if file.file.FileName == file2.file.FileName {
+				f1, err := io.ReadAll(file.file.Body)
+				if err != nil {
+					return nil, fmt.Errorf("error reading file: %w", err)
+				}
+				f2, err := io.ReadAll(file2.file.Body)
+				if err != nil {
+					return nil, fmt.Errorf("error reading file: %w", err)
+				}
+				file.file.Body = bytes.NewReader(f1)
+				file2.file.Body = bytes.NewReader(f2)
+				if bytes.Equal(f1, f2) {
+					foundDuplicated = true
+					filesGroup[group] = append(filesGroup[group], file)
+					break
+				}
+			}
+		}
+		if !foundDuplicated {
+			filesGroup = append(filesGroup, []*fileVariable{file})
+		}
+	}
+	if len(filesGroup) > 0 {
+		variablesString := []string{}
+		for i, files := range filesGroup {
+			variablesString = append(variablesString, fmt.Sprintf("\"%d\":[%s]", i, joinMapKeys(files)))
+		}
+		mapData = `{` + strings.Join(variablesString, ",") + `}`
+	}
+	err = bodyWriter.WriteField("map", mapData)
+	if err != nil {
+		return nil, fmt.Errorf("error writing map data to body: %w", err)
+	}
+
+	// files
+	for i, file := range filesGroup {
+		h := make(textproto.MIMEHeader)
+		dispParams := map[string]string{"name": strconv.Itoa(i)}
+		fileName := strings.TrimSpace(file[0].file.FileName)
+		if fileName != "" {
+			dispParams["filename"] = fileName
+		}
+		h.Set("Content-Disposition", mime.FormatMediaType("form-data", dispParams))
+		b, err := io.ReadAll(file[0].file.Body)
+		if err != nil {
+			return nil, fmt.Errorf("error reading file: %w", err)
+		}
+		h.Set("Content-Type", http.DetectContentType(b))
+		ff, err := bodyWriter.CreatePart(h)
+		if err != nil {
+			return nil, fmt.Errorf("error create multipart header: %w", err)
+		}
+		_, err = ff.Write(b)
+		if err != nil {
+			return nil, fmt.Errorf("error writing file to body: %w", err)
+		}
+	}
+	httpRequest.Body = io.NopCloser(bodyBuf)
+	httpRequest.Header.Set("Content-Type", bodyWriter.FormDataContentType())
+
+	return httpRequest, nil
+}
+
+func joinMapKeys(files []*fileVariable) string {
+	fileKeys := make([]string, len(files))
+	for i, v := range files {
+		fileKeys[i] = fmt.Sprintf("\"%s\"", v.mapKey)
+	}
+	return strings.Join(fileKeys, ",")
+}
```

Comment:
This function is trivial and used only once by `createUploadFileRequest`, so it's better to declare it as a literal function to avoid other functions from using it.

Reply:
fix